### PR TITLE
[OSSM-3269] Adding postsubmit jobs

### PR DESCRIPTION
**Pull request by leoieggli**

### CODE REVIEW

**Comment by yxun** [comment]
Are those pre-submit jobs ? It looks like they are not post submit ones, right ?

File: ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml [hunk 1]
```diff
@@ -76,6 +76,30 @@ tests:
...
+      - name: CI
+      - name: LOCAL_CPU_RESOURCES
+      - name: LOCAL_RAM_RESOURCES
+      - name: LOCAL_JOBS
+      from: src
+      resources:
+        requests:
+          cpu: 100m
+          memory: 100Mi
+      timeout: 8h0m0s
```
Comment (jwendell) [inline ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml:102]: need `postsubmit: true`

File: ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml [hunk 2]
```diff
@@ -76,6 +76,30 @@ tests:
...
+        ./tools/automator-main.sh \
+        -o maistra \
+        -f /creds-github/github-token \
+        -r proxy \
+        -c make maistra-update-everything \
+        '-t Automator: Update Envoy and dependencies' \
+        -l auto-merge \
+        -m bump-envoy
+      env:
+      - name: MAISTRA_NAMESPACE
```
Comment (jwendell) [inline ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml:91]: those vars are not needed

File: ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml [hunk 3]
```diff
@@ -76,6 +76,30 @@ tests:
           cpu: 100m
           memory: 100Mi
       timeout: 8h0m0s
+    - as: envoy-update-proxy-2.2
+      cli: latest
+      commands: |
+        ./tools/automator-main.sh \
+        -o maistra \
+        -f /creds-github/github-token \
```
Comment (jwendell) [inline ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml:84]: for this cred to work you will need something like https://github.com/openshift/release/blob/master/ci-operator/config/maistra/test-infra/maistra-test-infra-main.yaml#L42-L44

**Review by jwendell — CHANGES_REQUESTED** [review]
run `make update` and commit the files in the jobs dir

File: ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml (hunk 1 above)
Comment (leoieggli) [reply ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml]: Got it - fixed.

File: ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml (hunk 2 above)
Comment (leoieggli) [reply ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml]: Removed.

File: ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml (hunk 3 above)
Comment (leoieggli) [reply ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml]: Done.

File: ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml [hunk 4]
```diff
@@ -78,6 +78,22 @@ tests:
       timeout: 8h0m0s
     workflow: servicemesh-envoy-e2e-aws
   timeout: 8h0m0s
+- as: envoy-update-proxy-2.2
+  commands: |
+    ./tools/automator-main.sh \
```
Comment (yxun) [inline ci-operator/config/maistra/envoy/maistra-envoy-maistra-2.2.yaml:83]: where is the tools/automator-main.sh ?
https://github.com/maistra/envoy/tree/maistra-2.4/tools

This job container from: src means it starts from https://github.com/maistra/envoy

**Comment by yxun** [comment]
I think this postsubmit jobs PR should not target maistra/envoy repo. It should be target maistra/test-infra repo.
Then it will be related to https://issues.redhat.com/browse/OSSM-3666

**Review by yxun — CHANGES_REQUESTED** [review]
Do we know where should we have the ./tools/automator-main.sh  ? and when should it run ? Does this PR need to watch a maistra/envoy PR merged and then run the script from maistra/test-infra repo ?

**Review by jwendell — COMMENTED** [review]
We need to also clone the test-infra repo, similar to https://github.com/maistra/test-infra/blob/894de6e19fa286b4105dc0a25df523ff074414fb/prow/config/postsubmits.yaml#L247-L251

**Pull request closed without merging**